### ISSUE TITLE
BelongsToSelect - All of the related records are loaded

### ISSUE THREAD

**Issue by danielbidala** [laravel-filament/filament#253]
First of all thank you guys for Filament. It's extreamly powerfull.

The documentation of `BelongsToSelect` says: "By default, options will only load when you start typing a search". This is not the case. All of the related records are loaded on pageload (thousands in my case).

When you type to search, it doesn't send ajax request to the server.

**Context**
- Package version: v1.65

**Comment by danharrin** [comment]
Hey, could you paste your models' relationship configuration as well as the field code you've tried?

**Comment by danielbidala** [comment]
@danharrin The details are below:

**Gauge model relation:**

```
    public function events()
    {
        return $this->hasMany(Event::class);
    }

```

**Events model relation:**

```
    public function gauge()
    {
        return $this->belongsTo(Gauge::class);
    }
```

**EventResource.php's field:**

```
     Components\BelongsToSelect::make('gauge_id')
            ->label('Mérőeszköz')
            ->relationship('gauge', 'unique_id'),
```

**Comment by tomheno** [comment]
Hi !

I made some tests on this : it seems related to https://github.com/laravel-filament/filament/commit/bd7e3817e35af0e912b3066f20c7c946c58d1f6c

The preload seems to happen by default (without `->preload()` specified), due to 

```
$this->getOptionsUsing(function () use ($callback) {
                $relationship = $this->getRelationship();

                $query = $relationship->getRelated();

                if ($callback) {
                    $query = $callback($query);
                }

                return $query
                    ->pluck($this->getDisplayColumnName(), $relationship->getOwnerKeyName())
                    ->toArray();
            });
```
            
 I disabled this behavior using `->getOptionsUsing(false)` on `BelongsToSelect`.

**Comment by danharrin** [comment]
Fixed by #268.

**Comment by danielbidala** [comment]
@tomheno Thanks for your efforts
@danharrin Thanks for the fix
